### DISCUSSION TITLE
Provide a anchor element as prop so that popover can be decoupled from anchor element

### DISCUSSION THREAD

**Discussion by agupta1989** [chakra-ui/chakra-ui#8840] [Ideas]
### Description

Creating list with popover creates multiple instances of popover. 

### Problem Statement/Justification

Can we create Single popover and provide the anchorElement from outside (perhaps via prop) so that same popover can be reused ? If I render a list with button, where button is trigger / anchor element, it will create those many number of popover in DOM. If using popover under portal then those many portal and popover instance gets attached into DOM. Is there a better way to handle this ?

![image](https://github.com/chakra-ui/chakra-ui/assets/3787965/c7b15693-9c27-4c1b-a060-b707ed4986b8)

### Proposed Solution or API

Like Material UI example, can we have `anchorEl` support where user can dynamically supply the anchorElement to popover and popover gets attached to it and render the same popover ?
Reference:
https://mui.com/material-ui/react-popper/#basic-popper

**Comment by tsuharesu** [comment]
I'm facing the same. I have a user created list with the potential of dozens of items, and having one popover for each element seems like wasteful usage of resources. I could use `isLazy` but I don't know if that would help at all.

**Comment by kazimirchuk** [comment]
For anybody struggling with this I think I found a workaround which works for some cases (not for me though):

If you have a reference to an element you want to become an anchor to it means you also have that element's offsets.

1. Create a controlled popover
2. In the perfect World we wouldn't have a trigger but we must render it because otherwise this whole approach won't work
3. On click of an element you have reference to save its `clientX` and `clientY` offsets and propagate them to the popover in question (while also flipping its `isOpen` state)
4. Inside the popover use the offset in positioning like so:
```
<PopoverRoot
      open={isOpen}
      positioning={{
        placement: isBaseSize ? "top" : "right",
        flip: true, // doesn't work... 
        offset: {
          mainAxis: offsetX || 0,
          crossAxis: offsetY || 0,
        }
      }}
      onOpenChange={({ open: newOpen }) => {
        if (!newOpen) {
          onClose();
        }
      }}
>
```

Unfortunately this approach has one major downside. It can't automatically flip the popover from `"right"` to `"left"` because apparently if we specify offset then flipping doesn't work. I thought I could calculate the placement dynamically but if the popover is shown then no matter what the placement doesn't change...

So unless you need flipping to work this might be a workaround.

P.S. Maybe I'm doing something wrong that breaks the flipping but I spent way too much time on making it work so I don't even care at this point. The Popover is raw and not working like it should for the library of this magnitude, especially considering that v2 worked like a charm.

**Comment by segunadebayo** [comment]
Put `value` on each `Popover.Trigger` under one `Root`. The same popover moves to the active trigger.

https://chakra-ui.com/docs/components/popover#multiple-triggers

If the target is not a button, use `positioning.getAnchorRect`.
